test/e2e/networkpolicy_test.go

```diff
@@ -806,6 +806,107 @@ func TestIngressPolicyWithoutPortNumber(t *testing.T) {
 	}
 }
 
+func TestIngressPolicyWithEndPort(t *testing.T) {
+	skipIfHasWindowsNodes(t)
+
+	data, err := setupTest(t)
+	if err != nil {
+		t.Fatalf("Error when setting up test: %v", err)
+	}
+	defer teardownTest(t, data)
+
+	serverPort := int32(80)
+	serverEndPort := int32(84)
+	policyPort := int32(81)
+	policyEndPort := int32(83)
+
+	var serverPorts []int32
+	for i := serverPort; i <= serverEndPort; i++ {
+		serverPorts = append(serverPorts, i)
+	}
+
+	serverName, serverIPs, cleanupFunc := createAndWaitForPodWithMultiPorts(t, data, data.createAgnhostPodOnNodeWithMultiPort, "test-server-", "", serverPorts)
```

Comment:
instead of defining so many new functions, I think you can define `createAgnhostPodOnNodeWithMultiPort` here as a lambda function and then use `createAndWaitForPod` normally:

```go
createAgnhostPodOnNodeWithMultiPort := func() {
    ...
    // refer to serverPorts directly here
}
serverName, serverIPs, cleanupFunc := createAndWaitForPod(t, data, createAgnhostPodOnNodeWithMultiPort, "test-server-", "")
```

Reply:
Done.